src/injected/content/index.js

```diff
@@ -106,68 +132,88 @@ function checkInjectable() {
     a.id = domId;
     document.documentElement.appendChild(a);
   };
-  inject(`(${detect.toString()})(${JSON.stringify(id)})`);
+  inject(`(${detect})(${jsonDump(id)})`);
   const a = document.querySelector(`#${id}`);
   const injectable = !!a;
   if (a) a.parentNode.removeChild(a);
   return injectable;
 }
 
 function injectScripts(contentId, webId, data, scriptLists) {
-  const props = {};
-  [
-    Object.getOwnPropertyNames(window),
-    Object.getOwnPropertyNames(global),
-  ].forEach((keys) => {
-    keys.forEach((key) => { props[key] = 1; });
-  });
+  let props = [];
+  // combining directly to avoid GC due to a big intermediate object
+  const addUniqProp = key => !props.includes(key) && props.push(key);
+  const windowProps = Object.getOwnPropertyNames(window);
+  // buggy Firefox may list duplicate props
+  if (isFirefox) windowProps.forEach(addUniqProp);
+  else props = windowProps;
```

Comment:
What if other browsers give duplicate props too? I think we'd better just use `addUniqProp`.